src/main/java/io/appium/java_client/flutter/commands/ScrollParameter.java

```diff
@@ -56,13 +56,13 @@ public Map<String, Object> toJson() {
 
         params.put("finder", parseFlutterLocator(scrollTo));
         Optional.ofNullable(scrollView)
-                .ifPresent(scrollView -> params.put("scrollView", scrollView));
+                .ifPresent(scrollView -> params.put("scrollView", parseFlutterLocator(scrollView)));
         Optional.ofNullable(delta)
                 .ifPresent(delta -> params.put("delta", delta));
         Optional.ofNullable(maxScrolls)
-                .ifPresent(maxScrolls -> params.put("delta", maxScrolls));
+                .ifPresent(maxScrolls -> params.put("maxScrolls", maxScrolls));
```

Comment:
technically this is a breaking change. Are you sure it would work with older server versions?

Reply:
Yes it will work will older versions too. The issue is, at line 61 the 'delta' key again gets override at line 63. It means that maxScrolls was never set.